feeds/src/state/ducks/aggregator/selectors.ts

```diff
@@ -3,16 +3,54 @@ import { createSelector } from 'reselect'
 import { AppState } from 'state'
 import { OracleNode } from '../../../config'
 
-const upcaseOracles = (
-  state: AppState,
+type OracleAddress = string
+type NodeAddress = string
+
+type StateBranch = {
+  aggregator: {
+    oracleNodes: {
+      [address: string]: {
+        name: string
+        nodeAddress: NodeAddress[]
+        oracleAddress: OracleAddress
+        [prop: string]: any
+      }
+    }
+    config: {
+      contractVersion: number
+      [prop: string]: any
+    }
+  }
+  [prop: string]: any
+}
+
+export const upcaseOracles = (
+  state: StateBranch,
 ): Record<OracleNode['address'], OracleNode['name']> => {
-  return Object.fromEntries(
-    Object.keys(state.aggregator.oracleNodes).map(k => [
-      k.toUpperCase(),
-      state.aggregator.oracleNodes[k].name,
-    ]),
-  )
+  /**
+   * In v2 of the contract, oracles' list has oracle addresses,
+   * but in v3 - node addresses. Therefore, a different record of
+   * pairs has to be made for each contract version.
+   */
+  if (state.aggregator.config.contractVersion === 2) {
+    return Object.fromEntries(
+      Object.entries(
+        state.aggregator.oracleNodes,
+      ).map(([oracleAddress, oracleNodeObject]) => [
```

Comment:
How do you feel about renaming `oracleNodeObject` -> `oracleNode`?

Reply:
I feel very good about it - `Object` is redundant 😅